src/modules/treeland-personalization-manager/personalizationmanager.cpp

```diff
@@ -161,52 +162,56 @@ void QuickPersonalizationManager::onCursorContextCreated(personalization_cursor_
             &QuickPersonalizationManager::onGetCursorSize);
 }
 
-QString QuickPersonalizationManager::saveImage(personalization_wallpaper_context_v1 *context,
+void QuickPersonalizationManager::writeContext(personalization_wallpaper_context_v1 *context, const QByteArray &data, const QString &dest)
+{
+    QFile dest_file(dest);
+    if (dest_file.open(QIODevice::WriteOnly)) {
+        dest_file.write(data);
+        dest_file.close();
+
+        saveWallpaperSettings(dest, context);
+        Q_EMIT backgroundChanged(context->output_name, context->isdark);
+    }
+}
+
+void QuickPersonalizationManager::saveImage(personalization_wallpaper_context_v1 *context,
                                                const QString prefix)
 {
     if (!context || context->fd == -1 || context->output_name.isEmpty())
-        return QString();
+        return;
 
     QFile src_file;
     if (!src_file.open(context->fd, QIODevice::ReadOnly))
-        return QString();
+        return;
 
     QByteArray data = src_file.readAll();
     src_file.close();
 
-    QString dest_path = m_cacheDirectory + prefix + "_" + context->output_name;
-
     QDir dir(m_cacheDirectory);
     if (!dir.exists()) {
         dir.mkpath(m_cacheDirectory);
     }
 
-    QFile dest_file(dest_path);
-    if (dest_file.open(QIODevice::WriteOnly)) {
-        dest_file.write(data);
-        dest_file.close();
-
-        saveWallpaperSettings(dest_path, context);
-        return dest_path;
+    QString dest = m_cacheDirectory + prefix + "_" + context->output_name;
+    if (context->output_name.isEmpty()) {
+        Q_FOREACH (QScreen *screen, QGuiApplication::screens()) {
```

Comment:
Note: Since Qt 5.7, the use of this macro is discouraged. Use C++11 range-based for, possibly with std::as_const(), as needed.

Reply:
for (QScreen *screen : QGuiApplication::screens())